streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -838,6 +838,7 @@
     static {
         final Map<String, Object> tempProducerDefaultOverrides = new HashMap<>();
         tempProducerDefaultOverrides.put(ProducerConfig.LINGER_MS_CONFIG, "100");
+        tempProducerDefaultOverrides.put(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG, 10000);
```

Comment:
nit: Could you add the original comment explaining why we set it to smaller value too?